### PR TITLE
+pre-commit

### LINKED ISSUE #383: +pre-commit
Formulae: https://formulae.brew.sh/formula/pre-commit

### PR DESCRIPTION
**Pull request by ddobrinskiy**
* +pre-commit
* resolves #383

### CODE REVIEW

**Comment by ddobrinskiy** [comment]
This is an experiment: when trying to build pre-commit, I get weird cython errors at build time and need to think of which dependencies to include.

What if we just used mamba as a build dependency?

It usually handles all compilations et cetera pretty well, and may be a candidate to simplify build scripts for python.

The downside is that mamba distro may be pretty heavy.

p.s. all this is pending merge of #388

**Comment by ddobrinskiy** [comment]
@mxcl ready for review

File: projects/pre-commit.com/package.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+distributable:
+  url: https://github.com/pre-commit/pre-commit/archive/refs/tags/v{{ version }}.tar.gz
+  strip-components: 1
+
+# if there’s a github then we can parse the versions
```
Comment (mxcl) [inline projects/pre-commit.com/package.yml]: we should remove the generated comments.

File: projects/pre-commit.com/package.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  github: pre-commit/pre-commit/releases/tags  # reads github *releases* but uses the tags of those releases
+  strip: /^v/
+
+dependencies:
+  python.org: '>=3.8'
+
+build:
+  dependencies:
+    pip.pypa.io: '*'
+    gnu.org/bash: '>=2'  # for our venv/relocation scripts: feel free to fix that
```
Comment (mxcl) [inline projects/pre-commit.com/package.yml]: actually no longer required for the generated script

File: projects/pre-commit.com/package.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    GIT_COMMITTER_EMAIL : "test@example.com"
+  script: |
+    test "$(pre-commit --version)" = "pre-commit {{version}}"
+    git config --global init.defaultBranch main # get rid of the warning
+    git init
+    mv $FIXTURE .pre-commit-config.yaml
+    pre-commit install
+    echo 'hi\n' > f.txt
+    git add f.txt
+    git commit f.txt -m "test"
```
Comment (mxcl) [inline projects/pre-commit.com/package.yml:37]: great test 🥰

**Review by mxcl — CHANGES_REQUESTED** [review]
A few cleanups

File: projects/pre-commit.com/package.yml (hunk 3 above)
Comment (ddobrinskiy) [reply projects/pre-commit.com/package.yml:37]: Shamelessly stolen from brew formula :laughing: 

https://github.com/Homebrew/homebrew-core/blob/24b557524b975cd7c5f458c147a1abcc9f511acf/Formula/pre-commit.rb

Comment (ddobrinskiy) [reply projects/pre-commit.com/package.yml:37]: Credit goes to @pharaujo

File: projects/pre-commit.com/package.yml (hunk 2 above)
Comment (ddobrinskiy) [reply projects/pre-commit.com/package.yml]: removed bash, pip still needed (could not build without it)

File: projects/pre-commit.com/package.yml (hunk 1 above)
Comment (ddobrinskiy) [reply projects/pre-commit.com/package.yml]: done

**Comment by ddobrinskiy** [comment]
> A few cleanups

all done

**Comment by ddobrinskiy** [comment]
> A few cleanups

@mxcl all done

**Review by mxcl — APPROVED** [review]

**Pull request merged**